bundle-size/github-utils.js

```diff
@@ -137,6 +141,80 @@ class GitHubUtils {
     const member = members[Math.floor(Math.random() * members.length)];
     return member.login;
   }
+
+  /**
+   * Convert a team slug to a team id.
+   *
+   * @param {string} teamName the team slug (`organization/team`).
+   * @return {number} the team id.
+   */
+  async getTeamId_(teamName) {
+    const cacheKey = `${CACHE_TEAM_ID_KEY}/${teamName}`;
+    const [org, teamSlug] = teamName.split('/');
+    let teamId = this.cache.get(cacheKey);
+    if (!teamId) {
+      this.log(`Cache miss for ${cacheKey}. Fetching from GitHub...`);
+      teamId = await this.github.teams
+        .getByName({org, team_slug: teamSlug})
+        .then(result => result.data.id);
+      this.log(
+        `Fetched team id ${teamId} for team ${teamName} from GitHub. Caching ` +
+          `for ${CACHE_TEAM_ID_TTL_SECONDS} seconds.`
+      );
+      this.cache.set(cacheKey, teamId, CACHE_TEAM_ID_TTL_SECONDS);
+    }
+    return teamId;
+  }
+
+  /**
+   * Get the list of all team members by team id.
+   *
+   * @param {number} teamId the team id.
+   * @return {!Array<string>} the team id.
+   */
+  async getTeamMembers_(teamId) {
+    const cacheKey = `${CACHE_TEAM_MEMBERS_KEY}/${teamId}`;
+    let teamMembers = this.cache.get(cacheKey);
+    if (!teamMembers) {
+      this.log(`Cache miss for ${cacheKey}. Fetching from GitHub...`);
+      teamMembers = await this.github.teams
+        .listMembers({team_id: teamId})
+        .then(result => result.data.map(user => user.login));
+      this.log(
+        `Fetched team members [${teamMembers.join(', ')}] for team id ` +
+          `${teamId}. Caching for ${CACHE_TEAM_MEMBERS_TTL_SECONDS} seconds.`
+      );
+      this.cache.set(cacheKey, teamMembers, CACHE_TEAM_MEMBERS_TTL_SECONDS);
+    }
+    return teamMembers;
+  }
+
+  /**
+   * Choose a random reviewer from list of potential approver teams.
+   *
+   * @param {!Array<string>} potentialApproverTeams potential teams that can
+   *   approve this pull request.
+   * @param {number} pullRequestId the pull request id.
+   * @return {string} the chosen reviewer username.
+   */
+  async getRandomReviewer(potentialApproverTeams, pullRequestId) {
+    // First, randomly choose a team.
```

Comment:
I don't think this two-tiered approach is desirable. Imagine a case where we have `wg-viewers` (2 members) and `wg-caching` (12 members). Members of `wg-viewers` will get way more reviews than members of `wg-caching`. Since team member lists are cached, it shouldn't be particularly expensive to just get the full set of all members of all 2-3 teams and choose randomly from there

Reply:
Makes sense. PTAL at latest commit

Reply:
Discussed offline. Merging.